src/charts/brush.js

```diff
@@ -60,55 +60,45 @@ export default function Brush (_container) {
     cache.chartBrush.selectAll(".brush-rect")
       .attr("height", config.chartHeight)
 
-    moveBrush()
+    setBrush()
   }
 
-  function getDataExtent () {
+  function getDataExtentUnderBrush () {
     const selection = d3.event.selection
-    const dataExtent = selection.map((d) => invertScale(scales.xScale, d, config.keyType))
-    return dataExtent
-  }
-
-  function clampBrush (_dataExtent) {
-    if (config.keyType === "time") {
-      return [
-        new Date(clamp(new Date(config.brushRangeMin), _dataExtent)),
-        new Date(clamp(new Date(config.brushRangeMax), _dataExtent))
-      ]
-    } else {
-      return [
-        clamp(config.brushRangeMin, _dataExtent),
-        clamp(config.brushRangeMax, _dataExtent)
-      ]
-    }
+    return selection.map((d) => invertScale(scales.xScale, d, config.keyType))
   }
 
-  function moveBrush () {
-    const dataExtent = scales.xScale.domain()
-    const extent = clampBrush(dataExtent)
+  function setBrush () {
+    let extent = [config.brushRangeMin, config.brushRangeMax]
     if (extendIsValid(extent)) {
+      if (config.keyType === "time") {
+        extent = extent.map(d => new Date(d))
+      }
       cache.root.call(cache.brush.move, extent.map((d) => scales.xScale(d)))
     }
     return this
   }
 
   function handleBrushStart () {
-    if (!d3.event.sourceEvent || !d3.event.selection) {
+    if (!d3.event.sourceEvent ||
```

Comment:
This is the trick to prevent dispatching event unless it really comes from a user event